File: practice20/2.cpp
//https://www.nowcoder.com/practice/42852fd7045c442192fa89404ab42e92?tpId=98&&tqId=33006&rp=7&ru=/activity/oj&qru=/ta/2019test/question-ranking
#include<iostream>
#include<string>
using namespace std;
int main()
{
	int i, j, n, k;
	while (cin >> n)
	{
		string *pt = new string[n];
		for (i = 0; i < n; i++)
		{
			cin >> pt[i];
		}
		for (i = 0; i < n; i++)
		{
			for (j = 0; j < pt[i].size(); j++)
			{
				k = j;
				while (pt[i][j] == pt[i][j + 1] && pt[i][j + 1] == pt[i][j + 2])
				{
					pt[i].erase(j, 1);
				}
				while (k <= pt[i].size() - 4 && pt[i][j] == pt[i][j + 1] && pt[i][j + 2] == pt[i][j + 3])
				{
					pt[i].erase(j + 2, 1);
				}
			}
		}
		for (i = 0; i < n; i++)
		{
			cout << pt[i] << endl;
		}
	}
	system("pause");
	return 0;
}

File: practice30/2.java

//https://www.nowcoder.com/practice/d2dfc62bf1ba42679a0e358c57da9828?tpId=90&&tqId=30778&rp=1&ru=/activity/oj&qru=/ta/2018test/question-ranking
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String args[]){
        Scanner scanner=new Scanner(System.in);
        int n=scanner.nextInt();
        int []h=new int[n];
        for(int i=0;i<n;i++){
            h[i]=scanner.nextInt();
        }
        Arrays.sort(h);
        n=scanner.nextInt();
        int []w=new int[n];
        for(int i=0;i<n;i++){
            w[i]=scanner.nextInt();
        }
        Arrays.sort(w);
        int i=h.length-1;
        int j=w.length-1;
        int count=0;
        while(i>=0&&j>=0){
            if(w[j]>=h[i]){
                i--;
                j--;
                count++;
            }
            else{
                i--;
            }
        }
        System.out.println(count);
    }
}